cachito/workers/pkg_managers/gomod.py

```diff
@@ -187,7 +189,47 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
             )
             shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
 
-        return module, deps
+        # package level dependencies
+        go_list_deps_output = run_gomod_cmd(
+            ("go", "list", "-deps", "-f", "{{if not .Standard}}{{.ImportPath}} {{.Module}}{{end}}"),
+            run_params,
+        )
+
+        pkg_level_deps = []
+        for line in go_list_deps_output.splitlines():
+            # line example: pkg.io/foo/bar pkg.io/foo/var v1.0.0 => pkg.io/foo/bar v1.0.1
+            parts = [part for part in line.split(" ") if part not in ("")]
+            replaces = None
+
+            name = parts[0]
+            parent_module = parts[1]
+            if len(parts) > 2:
+                version = parts[2]
+            else:
+                # use requested module version
+                version = module_version
```

Comment:
Why is the version sometimes set and sometimes not? Is this the case where it's the actual package in the module?

If so, please add a comment in the code.

Reply:
Yes, these are the packages and sub-packags contained in the current module. I'll include a better comment here